inference-engine/src/mkldnn_plugin/nodes/mkldnn_convert_node.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_convert_node.h"
+#include "common/cpu_convert.h"
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+MKLDNNConvertNode::MKLDNNConvertNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
+        MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNConvertNode::getSupportedDescriptors() {
+    if (outDims.empty() && output.getLayout() != InferenceEngine::Layout::ANY)
+        outDims.push_back(MKLDNNDims(output.getDims()));
+    if (inDims.empty() && input.getLayout() != InferenceEngine::Layout::ANY)
+        inDims.push_back(MKLDNNDims(input.getDims()));
+    if (getParentEdges().size() != 1)
+        THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
+    if (getChildEdges().empty())
+        THROW_IE_EXCEPTION << "Incorrect number of output edges for layer " << getName();
+}
+
+void MKLDNNConvertNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    auto layer = getCnnLayer();
+    if (layer == nullptr) {
+        THROW_IE_EXCEPTION << "Cannot get CNN layer for layer name " << getName();
+    }
+
+    LayerConfig config;
+    DataConfig dataIn;
+    DataConfig dataConfigOut;
+
+    memory::format fmt;
+
+    if (input.getLayout() != InferenceEngine::Layout::ANY && output.getLayout() != InferenceEngine::Layout::ANY) {
```

Comment:
Not sure this is very safe check. If smb will decide to change default TesnorDesc ctor the code block will be invalid.

Reply:
I fully agree with you. References to the input/output descriptors have been changed to pointers so that their null state can be checked. (see commit e1639f3).